### PR TITLE
ROR capstone project 'Budget App'

### PR DESCRIPTION
**Pull request by mohamedgamaleldin999999**
Description: 💬
This pull request for illustrate what I've done to finish the budget mobile app using RoR

Done: ✅
 Create a rails project

 Setup the linters, rubocop & styleint

 Setup the devise for authentication

 Added the navbar

 Setup the home page

 Setup bootstrap

 Create groups table

 Create deal table

 Adding the references between tables

 Create the data models for deals and groups, and set up the relationships between them.

 Create the controllers and views for deals and groups, and set up the routes.

 Add the deals and groups index and new views.

 Add the create, index, and show actions to the deals and groups controllers.

 Update the deals and groups tables to add the required columns and foreign keys.

 Add the splash controller and view, and set up the routes.

 Setup and Using Rspec for test

 Implement different test for different components to ensure it's consistency.

 Adding README

Documentation: [Presentation video](https://youtu.be/KgvMz6qQD24)

### CODE REVIEW

**Comment by Reem-lab** [comment]
- Could you check your slack?

File: README.md
Comment (Efiamotu-1) [inline README.md:1]: - [ ] Your readme is currently missing an important project requirement as shown in the image below, please kindly record this video and add it to your readme before resubmitting :+1: 
pr

File: README.md [hunk 1]
```diff
@@ -1 +1,203 @@
...
+<details>
+<summary>Bootstrap</summary>
+  <ul>
+    <li><a href="https://getbootstrap.com/">Bootstrap</a></li>
+  </ul>
+</details>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link]()
```
Comment (Efiamotu-1) [inline README.md]: - [ ] Your live demo is also missing and this is an important requirement for this project please kindly add it before resubmit :+1:

File: app/views/layouts/application.html.erb
Comment (Efiamotu-1) [inline app/views/layouts/application.html.erb:1]: - [ ] The button on the transaction page should read `Add New Transaction` not `new deal` please kindly correct this
 
tp

Comment (Efiamotu-1) [inline app/views/layouts/application.html.erb:1]: - [ ] You are missing this below requirement for your transactions page, please kindly fix it 
rew

File: app/controllers/deals_controller.rb
Comment (Efiamotu-1) [inline app/controllers/deals_controller.rb:1]: [OPTIONAL]
- [ ] In all of your controllers you didn't create methods to handle update and destry action, i kindly advise that you add them to your project

Comment (Efiamotu-1) [inline app/controllers/deals_controller.rb:1]: - [ ] YOu currently have two tests failing at the moment as shown in the image below, please kindly fix them :+1: :sparkles: 
fl

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @mohamedgamaleldin999999, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md
Comment (OyePriscilla) [reply README.md:1]: - [ ] Please kindly make a video presentation as suggested earlier in accordance to the project requirement. 🙏

File: spec/views/transactions_spec.rb
Comment (OyePriscilla) [inline spec/views/transactions_spec.rb:1]: Some of the tests written that were passing in previous commit are now failing, perhaps due to the new implementation done. Kindly fix this. 🙏 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @mohamedgamaleldin999999  👏 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Live demo work properly
Designed transaction page according to the given template

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mohamedgamaleldin999999** [comment]
Here's my required documentation:

[Presentation video](https://youtu.be/KgvMz6qQD24)

thanks in advance.

**Review by topeogunleye — APPROVED** [review]
Hi @mohamedgamaleldin999999, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

> ## (Highlights) Good Points: 💪 
> * Added presentation video according to project requirements :heavy_check_mark:

çrpik-kurdî-impressive



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear
Please, remember to tag @topeogunleye in your question so I can receive the notification.
 
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**